roofit/roofitcore/src/RooVectorDataStore.cxx

```diff
@@ -1301,16 +1302,18 @@ void RooVectorDataStore::recalculateCache( const RooArgSet *projectedArgs, Int_t
 {
   if (!_cache) return ;
 
-  pRealVector tv[1000] ;
+  if (static_cast<Int_t>(_cacheUpdateList.size()) < _cache->_nReal) {
```

Comment:
I'd check for the capacity of the vector not the size and call reserve instead of resize.